code/game/objects/items/deployable/barricade.dm

```diff
@@ -0,0 +1,212 @@
+	var/drop_amount = 1
+
+/obj/structure/barricade/deconstruct(disassembled = TRUE)
+	if(!(flags_1 & NODECONSTRUCT_1))
+		make_debris()
+	qdel(src)
+
+/obj/structure/barricade/proc/make_debris()
+	return
+
+/obj/structure/barricade/attackby(obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_WELDER && user.a_intent != INTENT_HARM && bar_material == METAL)
+		if(obj_integrity < max_integrity)
+			if(!I.tool_start_check(user, amount=0))
+				return
+
+			to_chat(user, "<span class='notice'>You begin repairing [src]...</span>")
+			if(I.use_tool(src, user, 40, volume=40))
+				obj_integrity = CLAMP(obj_integrity + 20, 0, max_integrity)
+
+	else if(I.GetID() && initial(locked_down))
+		if(allowed(user))
+			locked_down = !locked_down
+			to_chat(user, "<span class='notice'>You [locked_down ? "lock" : "unlock"] the release mechanism.</span>")
+		else
+			to_chat(user, "<span class='warning'>Access denied.</span>")
+		return
+
+	else
+		return ..()
+
+/obj/structure/barricade/CanAllowThrough(atom/movable/mover, turf/target)//So bullets will fly over and stuff.
+	. = ..()
+	if(locate(/obj/structure/barricade) in get_turf(mover))
+		return TRUE
+	else if(istype(mover, /obj/item/projectile))
+		if(!anchored)
+			return TRUE
+		var/obj/item/projectile/proj = mover
+		if(proj.firer && Adjacent(proj.firer))
+			return TRUE
+		if(prob(proj_pass_rate))
+			return TRUE
+		return FALSE
+
+/obj/structure/barricade/MouseDrop(over_object, src_location, over_location)
+	. = ..()
+	if(over_object == usr && Adjacent(usr))
+		if(!ishuman(usr) || !usr.canUseTopic(src, BE_CLOSE))
+			return FALSE
+		if(!pickup_damaged && obj_integrity < max_integrity)
+			to_chat(usr, "<span class='warning'>[src] is damaged! You'll have to repair it before you can relocate it.</span>")
+			return FALSE
+		if(locked_down)
+			to_chat(usr, "<span class='warning'>[src] is still locked down! Swipe your ID to unlock it.</span>")
+			return FALSE
+
+		usr.visible_message("<span class='notice'>[usr] begins breaking down [src]</span>", "<span class='notice'>You begin breaking down [src].</span>")
+		if(do_after(usr, pickup_delay, src))
+
+			//If the barricade is made of parts, some of them are damaged when the barricade is damaged so we set how many are being returned here
+			if(initial(drop_amount) > 1)
+				drop_amount = round(drop_amount * (obj_integrity/max_integrity))
+			//If we are only picking up one item at most, it has a chance to fall apart based on damage the barricade accrued. Will always succeed if pickup_damaged is false.
+			else if(!prob(round((obj_integrity/max_integrity), 0.01) * 100))
+				usr.visible_message("<span class='notice'>[usr] tries to pick up [src] but it falls apart!</span>", "<span class='notice'>[src] is too damaged and falls apart!</span>")
+				qdel(src)
+				return FALSE
+
+			usr.visible_message("<span class='notice'>[usr] picks up [src].</span>", "<span class='notice'>You pick up [src].</span>")
+			qdel(src)
+			return TRUE
```

Comment:
I think rather than returning true/false you should have an overridable proc that creates the pickup item

Reply:
Done now 👍